### PR TITLE
Fix to new version of wallet (SQLite) code can run in cucumber

### PR DESCRIPTION
**Pull request by Jualns**
# Description

Many wallet functions are more optimized and faster now with SQLite, so several cucumber steps were no longer running.

For that I had to correct each scenario individually and making the necessary corrections.

The tests are working.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Chore (miscellaneous changes e.g. modifying .gitignore)
- [ ] Build (Affect build components like build tool, ci pipeline, dependencies, project version)
- [ ] Docs (documentation update)

# How Has This Been Tested?

Manually before release of the new version to epic-wallet (after the 3.0.0)

### CODE REVIEW

File: features/floonet.feature [hunk 1]
```diff
@@ -1,36 +1,36 @@
 Feature: There is a need to validate the floonet environment
 
-  #"/home/jualns/Desktop/epic/target/release/epic"
-  #"/home/ba/Desktop/EpicV3/epic-wallet/target/release/epic-wallet"
-  #"/home/ba/Desktop/epic-miner/target/debug/epic-miner"
+	#"/home/jualns/Desktop/epic/target/release/epic"
```
Comment (jhelison) [inline features/floonet.feature]: Remove this comments

File: features/scalability.feature [hunk 2]
```diff
@@ -1,40 +1,39 @@
...
-      #  Then I confirm all transactions
-      #  Then I check <txs> informations
-      #  Then I check <outputs> informations
-      #  Then I check <info> informations
\ No newline at end of file
+#@serial
+#Scenario: Testing the operation of a huge wallet
+#  Given I am using the <usernet> network
+#  And I start the local node
+#  And I initiate a wallet <Vitex_test> as w1
```
Comment (jhelison) [inline features/scalability.feature:33]: Why these comments?

File: features/transactions.feature [hunk 3]
```diff
@@ -1,57 +1,55 @@
...
+    Given I have a wallet with coins
+    When I send 0.001 coins with file method
+    And I receive the file transaction
+    And I finalize the file transaction
+    And I await confirm the transaction
+    Then I have 2 new transactions in txs
+    And I kill all running epic systems
+
+#Scenario:  Test Keybase/TOR send methods
+#  Given I have a testing chain
```
Comment (jhelison) [inline features/transactions.feature:51]: Why this comment?

File: features/wallet_operation.feature [hunk 4]
```diff
@@ -1,26 +1,30 @@
...
-    #  And I send 3 coins with http method
-    #  Given I initiate a miner
-    #  And I mine 15 blocks and stop miner
-    #  When I make a recovery
-    #  Then I have a wallet with 2 outputs transactions and 26 mined blocks
\ No newline at end of file
+#@serial
+#Scenario: Testing the operation of a new wallet
+#  Given I initiate a wallet
+#  And I initiate a miner
```
Comment (jhelison) [inline features/wallet_operation.feature:23]: Why this comment?

File: src/lib.rs [hunk 5]
```diff
@@ -45,35 +45,52 @@ pub fn str_to_chain_type(shortname: &str) -> ChainTypes {
...
+                .arg("--usernet")
+                .spawn()
+                .expect("failed to execute process"),
+            ChainTypes::Mainnet => Command::new(&binary_path)
+                .spawn()
+                .expect("failed to execute process"),
+            _ => panic!("Specified network does not exist!"),
+        },
     };
     // let output = if cfg!(target_os = "windows") {
```
Comment (jhelison) [inline src/lib.rs]: Remove comments

File: src/lib.rs [hunk 6]
```diff
@@ -91,59 +108,83 @@ pub fn spawn_network(chain_type: &ChainTypes, binary_path: &str, policy: &str) -
...
+            server_toml
+                .members
+                .as_mut()
+                .unwrap()
+                .server
+                .p2p_config
+                .seeding_type = Seeding::List; //Some("List".to_owned());
+
             // ip Floonet VM 15.229.31.27:23414
             //let ip_1 = get_ip(15, 229, 31, 27, 23414);
```
Comment (jhelison) [inline src/lib.rs]: Remove comments

File: src/lib.rs [hunk 7]
```diff
@@ -91,59 +108,83 @@ pub fn spawn_network(chain_type: &ChainTypes, binary_path: &str, policy: &str) -
...
             let ip_1 = get_ip_new("15.229.31.27:23414");
-            let mut vec_ip:Vec<PeerAddr> = Vec::new();
+            let mut vec_ip: Vec<PeerAddr> = Vec::new();
             vec_ip.push(ip_1);
             //let vec_ip = vec![ip_1];
```
Comment (jhelison) [inline src/lib.rs]: Remove comments

File: src/lib.rs [hunk 8]
```diff
@@ -223,30 +261,38 @@ pub fn create_wallet(chain_type: &ChainTypes, binary_path: &str, password: &str)
...
-    
+pub fn send_coins_smallest(
+    chain_type: &ChainTypes,
+    binary_path: &String,
+    method: String,
+    password: &String,
+    amount: String,
+    destination: &String,
+) -> Output {
     //let str_amount = f32::to_string(&amount);
```
Comment (jhelison) [inline src/lib.rs]: Remove comments

File: src/lib.rs [hunk 9]
```diff
@@ -305,60 +389,76 @@ pub fn send_coins_smallest(chain_type: &ChainTypes, binary_path: &String, method
...
-    
+pub fn confirm_transaction(chain_type: &ChainTypes, binary_path: &str, password: &str) {
     // time dependence
     let t0 = Instant::now();
     let two_minute = Duration::from_secs(120);
-    
+
     while t0.elapsed() < two_minute {
         let values_info = info_wallet(chain_type, binary_path, password);
         //if values_info[4] > 0.0 && values_info[5] > 0.0 {
```
Comment (jhelison) [inline src/lib.rs]: Remove comments

File: src/lib.rs [hunk 10]
```diff
@@ -371,61 +471,64 @@ pub fn get_chain_height(chain_type: &ChainTypes, binary_path: &str, password: &s
...
-        txs_str.matches("Confirmed").count() as u32 - 1]; // -1 because header of txs command have "Confirmed?"
+        txs_str.matches("Confirmed").count() as u32 - 1, // attempt to subtract with overflow (It will probably give an overflow error if you don't find any Confirmed and remove 1 so it's -1 at u32;)
+    ]; // -1 because header of txs command have "Confirmed?"
     sent_receive_coinbase
 }
 
 pub fn get_http_wallet(chain_type: &ChainTypes) -> String {
     // TODO get from wallet toml (api_listen_interface = "127.0.0.1")
     let ip = "127.0.0.1";
+    //http://127.0.0.1:23415
```
Comment (jhelison) [inline src/lib.rs]: Remove comments

File: src/lib.rs [hunk 11]
```diff
@@ -478,87 +605,92 @@ pub fn generate_response_file_name(sent_file_name: &String) -> String {
...
     let mut height_vec: Vec<i32> = Vec::new();
     for element in output_msg_vec {
-        if element.contains("Total"){
+        if element.contains("Total") {
             let all_splits = element.split("\n").collect::<Vec<&str>>();
             let b: i32 = all_splits[0].parse().unwrap();
             height_vec.push(b);
         }
     }
     //let height_vec = output_msg_vec[1].split("\n").collect::<Vec<&str>>;
```
Comment (jhelison) [inline src/lib.rs]: Remove comments

File: src/main.rs [hunk 12]
```diff
@@ -103,143 +79,139 @@ pub struct PackTransaction {
...
+    wait_for(30);
 
     let mut handles_vec = Vec::new();
 
+    // Multi-thread but with only 1 thread, because wallet can't build multiple transactions in the same time (yet)
     for _ in 0..1 {
         let method_to_sent = Arc::clone(&method_to_send);
         let pass = Arc::clone(&password);
         let chain_t = Arc::clone(&chain_type);
         //let server_bin = Arc::clone(&server_binary);
```
Comment (jhelison) [inline src/main.rs]: Remove comments

File: features/wallet_operation.feature (hunk 4 above)
Comment (Jualns) [reply features/wallet_operation.feature:23]: Scenarios that were already planned for this feature but are not urgent for this release, so this scenario not ready.

File: features/transactions.feature (hunk 3 above)
Comment (Jualns) [reply features/transactions.feature:51]: Scenarios that were already planned for this feature but are not urgent for this release, so this scenario not ready.

File: features/scalability.feature (hunk 2 above)
Comment (Jualns) [reply features/scalability.feature:33]: Scenarios that were already planned for this feature but are not urgent for this release, so this scenario not ready.

**Review by jhelison — APPROVED** [review]

**Pull request merged**